### PR TITLE
Custom CKEditor plugin for browsing mediafiles

### PR DESCRIPTION
**Pull request by FinnStutzenstein**

### CODE REVIEW

**Comment by emanuelschuetze** [comment]
It would be nice to select additional 100% width. Without height.

**Comment by FinnStutzenstein** [comment]
@emanuelschuetze Changed from width x height to scale in percentage. Please check again.

**Comment by emanuelschuetze** [comment]
Please fix travis and rebase.

File: openslides/mediafiles/static/js/mediafiles/image-plugin.js [hunk 1]
```diff
@@ -0,0 +1,202 @@
+(function () {
+
+'use strict';
+
+angular.module('OpenSlidesApp.mediafiles.image-plugin', [])
```
Comment (normanjaeckel) [inline openslides/mediafiles/static/js/mediafiles/image-plugin.js]: Please add dependencies. See other AngularJS modules of this app (it is not complete, I know).

File: openslides/mediafiles/static/templates/mediafiles/image-plugin.html [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        </div>
+        <div>
+          <input type="number" id="image-scale" class="cke_dialog_ui_input_text"> %
+        </div>
+        <div id="originalSizeText"></div>
+      </div>
+    </td>
+  </tr>
+</table>
+
```
Comment (normanjaeckel) [inline openslides/mediafiles/static/templates/mediafiles/image-plugin.html]: It this empty line to be removed?

File: openslides/mediafiles/static/js/mediafiles/image-plugin.js [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+
+.factory('ImageBrowserPlugin', [
+    '$http',
```
Comment (normanjaeckel) [inline openslides/mediafiles/static/js/mediafiles/image-plugin.js]: Please also care of dependency order.

File: openslides/mediafiles/static/js/mediafiles/image-plugin.js [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+                        },
+                    };
+                };
+            },
+        };
+    }
+])
+
+.run([
+    'Editor',
```
Comment (normanjaeckel) [inline openslides/mediafiles/static/js/mediafiles/image-plugin.js]: Here the same.

File: openslides/mediafiles/static/js/mediafiles/image-plugin.js [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+                        CKEDITOR.tools.imagebrowser = {};
+
+                        // Initialize this dialog, if it is opened.
+                        editor.on('dialogShow', function (event) {
+                            var dialog = event.data;
+                            if (dialog.getName() === 'imagebrowser-dialog') {
+                                CKEDITOR.dialog.getCurrent().disableButton('ok');
+
+                                // Load the main plugin template and paste it into the container
+                                $http.get('static/templates/mediafiles/image-plugin.html').then(function (success) {
```
Comment (normanjaeckel) [inline openslides/mediafiles/static/js/mediafiles/image-plugin.js]: Is it possible to use AngularJS template cache here?

Comment (FinnStutzenstein) [inline openslides/mediafiles/static/js/mediafiles/image-plugin.js]: Yes, great idea!

File: openslides/mediafiles/static/js/mediafiles/image-plugin.js (hunk 1 above)
Comment (FinnStutzenstein) [inline openslides/mediafiles/static/js/mediafiles/image-plugin.js]: Done.

File: openslides/mediafiles/static/templates/mediafiles/image-plugin.html (hunk 2 above)
Comment (FinnStutzenstein) [inline openslides/mediafiles/static/templates/mediafiles/image-plugin.html]: Done.

File: openslides/mediafiles/static/js/mediafiles/image-plugin.js (hunk 3 above)
Comment (FinnStutzenstein) [inline openslides/mediafiles/static/js/mediafiles/image-plugin.js]: What do you exactly mean by order? Do we have a preferred order? I always do the angular staff first, then the resources and then all other stuff needed.

**Pull request merged**